wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -123,153 +141,158 @@ protected String getProxiedUrl() {
+      stmt.execute("START TRANSACTION READ ONLY");
+      stmt.executeQuery("SELECT 1");
 
       final SQLException exception =
           assertThrows(SQLException.class, () -> conn.setReadOnly(false));
-      String currentConnectionId = queryInstanceId(conn);
+      String currentConnectionId = auroraUtil.queryInstanceId(conn);
       assertEquals(SqlState.ACTIVE_SQL_TRANSACTION.getState(), exception.getSQLState());
       assertEquals(readerConnectionId, currentConnectionId);
 
       stmt.execute("COMMIT");
 
       conn.setReadOnly(false);
-      currentConnectionId = queryInstanceId(conn);
+      currentConnectionId = auroraUtil.queryInstanceId(conn);
       assertEquals(writerConnectionId, currentConnectionId);
     }
   }
 
   @TestTemplate
-  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
-  // Tests use Aurora specific SQL to identify instance name
-  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
-  public void test_setReadOnlyFalseInTransaction_setAutocommitZero() throws SQLException {
+  public void test_setReadOnlyFalseInTransaction_setAutocommitFalse() throws SQLException {
     try (final Connection conn = DriverManager.getConnection(getUrl(), getProps())) {
 
-      final String writerConnectionId = queryInstanceId(conn);
+      final String writerConnectionId = auroraUtil.queryInstanceId(conn);
 
       conn.setReadOnly(true);
-      final String readerConnectionId = queryInstanceId(conn);
+      final String readerConnectionId = auroraUtil.queryInstanceId(conn);
       assertNotEquals(writerConnectionId, readerConnectionId);
 
       final Statement stmt = conn.createStatement();
-      stmt.execute("SET autocommit = 0");
+      conn.setAutoCommit(false);
       stmt.executeQuery(
           // TODO: can we replace it with something less database specific?
           "SELECT COUNT(*) FROM information_schema.tables");
 
       final SQLException exception =
           assertThrows(SQLException.class, () -> conn.setReadOnly(false));
-      String currentConnectionId = queryInstanceId(conn);
+      String currentConnectionId = auroraUtil.queryInstanceId(conn);
       assertEquals(SqlState.ACTIVE_SQL_TRANSACTION.getState(), exception.getSQLState());
       assertEquals(readerConnectionId, currentConnectionId);
 
       stmt.execute("COMMIT");
 
       conn.setReadOnly(false);
-      currentConnectionId = queryInstanceId(conn);
+      currentConnectionId = auroraUtil.queryInstanceId(conn);
       assertEquals(writerConnectionId, currentConnectionId);
     }
   }
 
+
   @TestTemplate
-  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
-  // Tests use Aurora specific SQL to identify instance name
-  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnDatabaseEngine({DatabaseEngine.MYSQL})
   public void test_setReadOnlyTrueInTransaction() throws SQLException {
     try (final Connection conn = DriverManager.getConnection(getUrl(), getProps())) {
+      final AuroraTestUtility auroraUtil =
```

Comment:
I think we can use the class auroraUtil variable instead of creating another one here